core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/types/UtilTests.kt

```diff
@@ -151,6 +154,19 @@ class UtilTests {
 
         guessValueType(sequenceOf(1, 2, listOf(1), emptySet<Any>())) shouldBe typeOf<Any>()
         guessValueType(sequenceOf(listOf(1), setOf(1.0, 2.0))) shouldBe typeOf<Collection<Number>>()
+
+        guessValueType(
+            sequenceOf(DataColumn.empty(), columnOf(1)),
+            allColsMakesRow = true,
+        ) shouldBe typeOf<DataRow<*>>()
+
+        guessValueType(
+            sequenceOf(columnOf("a"), columnOf(1)),
+            allColsMakesRow = true,
+        ) shouldBe typeOf<DataRow<*>>()
+        guessValueType(
```

Comment:
split two guessValueType?

Reply:
then it would need to be split in 4 because we also have `listifyValues: Boolean`. I think in this case it's good to have all logic of `values -> column type` of the entire library handled by one function `guessValueType()`. This makes it easier to debug and reason about.